jax/_src/numpy/lax_numpy.py

```diff
@@ -3334,6 +3334,115 @@ class _Ogrid(_IndexGrid):
 ogrid = _Ogrid()
 
 
+class _AxisConcat:
+  """Concatenates slices, scalars and array-like objects along a given axis."""
+
+  def _maybe_process_directive(self, key):
+    """Adjust parameters when detecting a directive."""
+    directive = key[0]
+    # two special cases: matrix directives
+    mats = ("r", "c")
+    matrix = -1 if str(directive) not in mats else mats.index(directive)
+    params = [self.axis, self.ndmin, self.trans1d, matrix]
+
+    if not isinstance(directive, str) or matrix != -1:
+      return params
```

Comment:
This seems strange that we'd accept any possible value here if the directive is not a string. Shouldn't we raise an error if it's not an expected value?